src/textual/_opacity.py

```diff
@@ -25,8 +25,8 @@ def _apply_opacity(
     from_rich_color = Color.from_rich_color
     from_color = Style.from_color
     blend = base_background.blend
-    for segment in segments:
-        text, style, _ = segment
+    styled_segments = cast("Iterable[tuple[str, Style, object]]", segments)
```

Comment:
The second value in the tuple is typed as optional, but we know from context that it never is None.

Doing the cast outside of the loop for efficiency.